stl/inc/xmemory

```diff
@@ -653,6 +681,13 @@ struct _Default_allocator_traits { // traits for std::allocator
         }
     }
 
+#if _HAS_CXX23
+    _NODISCARD_RAW_PTR_ALLOC constexpr static allocation_result<pointer, size_type> allocate_at_least(
+        _Alloc& _Al, _CRT_GUARDOVERFLOW const size_type _Count) {
+        return _Al.allocate_at_least(_Count);

```

Comment:
`_Default_allocator_traits` is a pure throughput optimization, since we know what `std::allocator` will do. (We carefully activate this only for the primary template. It is unusual that we spend so much code on this, but it turns out to be a significant improvement in practice.) Here, `std::allocator` does the default thing:

https://github.com/microsoft/STL/blob/994f94199c561a38a4a8e15584030160095bc25a/stl/inc/xmemory#L955-L958

Therefore we should hardcode `return {_Al.allocate(_Count), _Count};` and save a step.